### PR TITLE
Add SPDX header

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* Resolves: none
* Target version: main

### TODO

- [x] finish-up adding missing header
- [x] run spdx converter
- [x] add reuse GH action

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Auto loader check fails: would need help to understand why and how to fix.
https://github.com/nextcloud/richdocuments/actions/runs/8968932531/job/24629343575?pr=3664

File: composer/autoload.php [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?php
-
-// autoload.php @generated by Composer
+/**
```
Comment (juliushaertl) [inline composer/autoload.php:2]: Can we rather specify this through the .reuse/dep5 file? Those files are autogenerated by composer to have a static autoloader and the CI checks if those match the generated ones.

Comment (AndyScherzinger) [reply composer/autoload.php:2]: Yes, sure, especially when auto generated 👍

Comment (AndyScherzinger) [reply composer/autoload.php]: @juliushaertl DONE 👍

**Comment by juliushaertl** [comment]
Haven't seen this cypress failure before and main branch is passing, need to look into why that happens

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**